tests/algorithms/pytorch/test_pca_algo.py

```diff
@@ -0,0 +1,219 @@
+        # the noise is random so the data is not identical on every organization.
+        contents=test_linear_data_samples_pca,
+        dataset_keys=[numpy_datasets[0]],
+        clients=[network.clients[0]],
+        tmp_folder=session_dir,
+    )
+
+    test_data_nodes = [
+        TestDataNode(
+            network.msp_ids[0],
+            numpy_datasets[0],
+            linear_samples,
+            metric_keys=[mae_metric],
+        )
+    ]
+
+    return test_data_nodes
+
+
+@pytest.fixture(scope="session")
+def test_linear_data_samples_pca():
+    """Generates linear linked data for testing purposes. The train_linear_data_samples data and
+    test_linear_data_samples data are linked with the same weights as they fixed per the same seed.
+
+    Returns:
+        List[np.ndarray]: A one element list containing linear linked data.
+    """
+    return [
+        assets_factory.linear_data(
+            n_col=LINEAR_N_COL + LINEAR_N_TARGET,
+            n_samples=64,
+            weights_seed=42,
+            noise_seed=42,
+        )
+    ]
+
+
+@pytest.fixture(scope="session")
+def train_linear_data_samples_pca(network):
+    """Generates linear linked data for training purposes. The train_linear_data_samples data and
+    test_linear_data_samples data are linked with the same weights as they fixed per the same seed.
+
+    Args:
+        network (Network): Substra network from the configuration file.
+
+    Returns:
+        List[np.ndarray]: A list of linear data for each organization of the network.
+    """
+    cov_matrix = np.array([[10.0, 0.5, 0.0], [0.5, 1.0, 0.0], [0.0, 0.0, 0.1]])
+    mean = np.array([100.0, -20.0, 10.0])
+
+    result = []
+    np.random.seed(42)
+    for _ in range(network.n_organizations):
+        data = np.zeros((100, 4))
+        data[:, :3] = np.random.multivariate_normal(mean, cov_matrix, size=100)
+        data[:, 3] = np.random.normal(0.0, 1.0, size=100)
+        result.append(data)
+    return result
+
+
+@pytest.mark.substra
+@pytest.mark.slow
+def test_cp_performance(network, compute_plan, session_dir, train_linear_data_samples_pca):
+    """Check the weight initialization, aggregation and set weights.
+    The aggregation itself is tested at the strategy level, here we test
+    the pytorch layer.
+    """
+
+    data = np.concatenate([d[:, :LINEAR_N_COL] for d in train_linear_data_samples_pca])
+    cov = np.cov(data.T)
+    _, eig = np.linalg.eig(cov)
+    numpy_pca_eigen_values = eig.T[:2]
+    fed_pca_model = download_last_aggregate_model(network, session_dir, compute_plan)
+    fed_pca_eigen_values = fed_pca_model.avg_parameters_update[0]
+    numpy_pca_eigen_values = np.array([np.sign(eigen_v[0]) * eigen_v for eigen_v in numpy_pca_eigen_values])
+    fed_pca_eigen_values = np.array([np.sign(eigen_v[0]) * eigen_v for eigen_v in fed_pca_eigen_values])
+    np.testing.assert_allclose(numpy_pca_eigen_values, fed_pca_eigen_values, rtol=1e-2)
```

Comment:
To get smaller rtol, we would need more rounds, and it will takes more time. But we can push to rtol=1e-5 with like 10 rounds or something like that.
The question, is up to which precision you want to check the convergence? And we'll update accordingly.

Reply:
Test is passing with rtol=1e-5 now